templates.dist/form/ownerleadgeneratorform.php

```diff
@@ -91,37 +88,43 @@
 					}
 				?>
 
-				<div class="lead-lightbox lead-page-1">
-					<h2><?php echo esc_html__('Your contact details', 'onoffice-for-wp-websites'); ?></h2>
-					<p>
-						<?php echo implode('<br>', $addressValues); ?>
-					</p>
-				</div>
-
-				<div class="lead-lightbox lead-page-2">
-					<h2><?php echo esc_html__('Information about your property', 'onoffice-for-wp-websites'); ?></h2>
-					<p>
-						<?php echo implode('<br>', $estateValues); ?>
-					</p>
-					<p>
-						<?php echo implode('<br>', $miscValues); ?>
-					</p>
-					<p>
-						<div style="float:right">
-							<?php
-							$pForm->setGenericSetting('formId', 'leadgeneratorform');
-							include(ONOFFICE_PLUGIN_DIR.'/templates.dist/form/formsubmit.php');
-							?>
-						</div>
-					</p>
-				</div>
+<?php 
+				$pageTitle = [
+					1 => esc_html__('Page 1', 'onoffice-for-wp-websites'),
+					2 => esc_html__('Page 2', 'onoffice-for-wp-websites'),
+					3 => esc_html__('Page 3', 'onoffice-for-wp-websites'),
+					4 => esc_html__('Page 4', 'onoffice-for-wp-websites'),
+					5 => esc_html__('Page 5', 'onoffice-for-wp-websites'),
+					6 => esc_html__('Page 6', 'onoffice-for-wp-websites'),
+				];
+				$totalPages = count($addressValues);
+				foreach ($addressValues as $pageNumber => $fields): ?>
+					<div class="lead-lightbox lead-page-<?php echo $pageNumber; ?>">
```

Comment:
in case there are more pages than pageTitles, it might cause problems. We should maybe add here some fallback / checks so the title won't be empty.